### ISSUE TITLE
File the Yes/No/Not Acceptable widget

### ISSUE THREAD

**Issue by goneall** [OpenChain-Project/Online-Self-Certification-Web-App#5]
The back-end support a yes/no + not acceptable question type.  The HTML/JavaScript/JQuery needs to be updated to represent this type to the user.

Question 5.d can be updated to use this question type once implemented.

Following are the known steps to making this work:
  - Implement the attached widgent provided by Alexios
  - Get the JavaScript in certsurvey.js to record and report the N/A answers
  - Test the implementation
  - Update the question 5.d to use the new question type

**Comment by goneall** [comment]
Attached is the widget provided by Alexios Zavros

**Comment by forakshu** [comment]
@goneall : Few questions:
1) I dont see widget attachement in previous comment. Did you miss it?
2) I saw implementation in"certsurvey.js" and I see there is code to support N/A answers. If question type is set to "YES_NO_NA" that code will kick in and it also records and report n/a answer. I havent updated the question 5.d to use the new question type yet. I just tested it locally.
3) I am not sure intention is to use the current implementation for YES/NO/NA or is the widget provided by Alexios Zavros different and we need to use that.

**Comment by goneall** [comment]
[yes_no_na.zip](https://github.com/OpenChain-Project/Online-Self-Certification-Web-App/files/1151209/yes_no_na.zip)
@forakshu Just attached the file
It has been a while since I look at this, but I believe the intent is to replace the widget to improve the look/feel.  It is possible this was fixed earlier and I forgot to close the issue, but I just looked back through the commit records and didn't see the commit.

**Comment by saramaebee** [comment]
Hi, is this still an active issue? Or are there any particular open issues good for beginners?

**Comment by goneall** [comment]
@saramaebee I believe this would still be useful and a good first time project
@shanecoughlan Let me know if you agree/disagree

**Comment by shanecoughlan** [comment]
I concur it would be a great starting point 🙂

@saramaebee welcome. Carlos from FOSSA told me you would be dropping by to assist with our web app.

It would be super useful if you could do a general review of our code, and consider how we could make it more sustainable and easier for people to join in.

Because we are an ISO standard maintainer, one of our big questions for 2022 is “how do we make this service last ten years?”

**Comment by saramaebee** [comment]
Hi Shane,

So far, my main hold-up has been setting up the development environment. It's been a while since I've worked in Eclipse, and while I did some research into the Eclipse Facets, I had trouble finding some of the options for setting it up.

This has hampered my progress a bit; however, I did take a look at certsurvey.js and have an alright grasp of how the survey comes together. In terms of stabilizing this app for the next ten years, the main concern I see is how the survey is put together. While the manner is clever, it does seem a bit outdated, with a JS function generating the raw HTML for the survey.

I think that this could be simplified quite a bit through the use of modern front-end libraries such as React/Angular, but that may also just be internal bias as I’m far wiser to those libraries over jQuery.

That said, in terms of the actual change requested by this ticket, the change itself should be fairly simple to implement with the main blocker at the moment being that I can’t set up the development environment.

Do you or anyone who has worked on the project previously have any advice for getting this environment set up so I can play around with the survey and try to make those changes?

**Comment by goneall** [comment]
> I think that this could be simplified quite a bit through the use of modern front-end libraries such as React/Angular, but that may also just be internal bias as I’m far wiser to those libraries over jQuery.

I couldn't agree more that the front end could be simplified.  I'm more of a back-end developer so I'll defer to the expertise of those working on more front-end UI.  I've heard good things about React from other colleagues, but either framework would be simpler than the jQuery based environment currently in use.

I would like to retain the same interface to the back-end - which is all REST based, which I don't think would be a problem.

Perhaps we could discuss this on the call as well.

**Comment by shanecoughlan** [comment]
> Hi Shane,
> 
> I'd love to do exactly that! While I investigate this bug, I'll come up with a list of ways I think this project could be more inviting towards new developers!
> 
> Thank you for connecting with Carlos and putting this together, I'm super excited to help out!
> 
> Out of my own personal curiosity, is there any documentation on the ISO this supports? I'd love to read up on it as well if possible.
> 
> Thanks again, Sara

Yup! Here is an overview of the standard:
https://www.openchainproject.org/about

"The OpenChain Project maintains the International Standard for open source license compliance. This allows companies of all sizes and in all sectors to adopt the key requirements of a quality open source compliance program. This is an open standard and all parties are welcome to engage with our community, to share their knowledge, and to contribute to the future of our standard."

The actual standard:
https://www.openchainproject.org/contribute-to-the-standard

Basically we are trying to build trust in the supply chain by making sure everyone has the key requirements of a quality open source compliance program in place.
